### ISSUE TITLE
python packages as in non-ml-workspace images

### ISSUE THREAD

**Issue by chritter** [StatCan/kubeflow-containers-desktop#19]
Hello,
On the non ml-workspace images we have a variety of extensions enabled, including those to supporting Python styles. Could we have these extensions installed on the desktop images as well?

Thanks!

**Comment by blairdrummond** [comment]
What do you mean by "Python Styles" and what kinds of extensions? Just pip packages?

**Comment by chritter** [comment]
Yes, the Jupyter extensions, most importantly

* The code linting extension installed by @blairdrummond on the non-ml-workspace images
* The code commenting extension
* The cell execution time extension

Thanks!

**Comment by frazs** [comment]
I'm not sure what these are or where they are defined

**Comment by ca-scribner** [comment]
hey @blairdrummond / @chritter , do you know which extensions are missing (or is this still an issue)?  @saffaalvi is available to work on this but we weren't sure which extensions are needed

**Comment by blairdrummond** [comment]
Mostly referring to these;

https://github.com/StatCan/kubeflow-containers/blob/35745bc39e0c63fd3fbf29acbf165926ca1d09a1/minimal-notebook/cpu/Dockerfile#L52-L112

The commenting extension actually isn't in the main image yet, to the best of my knowledge. But it's also a great addition!

**Comment by blairdrummond** [comment]
I'm not sure which extensions are most important, but that's the global list

**Comment by ca-scribner** [comment]
Is there any reason we can’t have a shared base extension list for all our
core notebook images? They fee very bespoke atm

**Comment by frazs** [comment]
Wait a sec: it's my impression that we've decided to completely remove `jupyterlab` from `remote-desktop` (the images that used to be called `ml-workspace`), and were considering removing `jupyter` as well. These appear to be `jupyterlab` extensions...?

**Comment by chritter** [comment]
@frazs Please do not remove Jupyter from the ml-workspace image! It is a great capability we are using daily in the construction starts project. I like @ca-scribner to have a list of base extensions, which should include the linting support, debugging feature, git and more.

**Comment by frazs** [comment]
Closing this issue as the current direction is that JupyterLab and its extensions will not be in the official Remote Desktop images.

Solutions for sharing storage between different images are in the works (https://github.com/StatCan/daaas/issues/194 and https://github.com/StatCan/daaas/issues/133), such that in the future it should become possible to have specialized JupyterLab images accessing the same data alongside Remote Desktop images.

In the meantime, as we have discussed, I have made available an [experimental placeholder](https://github.com/frazs/kubeflow-containers-desktop-experimental/tree/master/remote-desktop-jupyterlab) version of Remote Desktop geomatics for your team that does include JupyterLab and its extensions. It will be decommissioned once multi-image shared storage is ready and confirmed working.
